common/views/components/FilterDrawerRefine/FilterDrawerRefine.js

```diff
@@ -147,9 +170,26 @@ const FilterDrawerRefine = ({
   changeHandler,
 }: Props) => {
   const workTypeInUrlArray = searchParams.workType || [];
-  const { productionDatesFrom, productionDatesTo, workType } = searchParams;
+  const { productionDatesFrom, productionDatesTo } = searchParams;
   const [inputDateFrom, setInputDateFrom] = useState(productionDatesFrom);
   const [inputDateTo, setInputDateTo] = useState(productionDatesTo);
+  const workTypeFilters = allWorkTypes
+    .map(workType => {
+      const matchingWorkTypeAggregation = workTypeAggregations.find(
+        ({ data }) => workType.data.id === data.id
+      );
+      const matchingAppliedWorkType = workTypeInUrlArray.find(
+        id => workType.data.id === id
+      );
+      if (matchingWorkTypeAggregation) {
+        return matchingWorkTypeAggregation;
+      } else if (matchingAppliedWorkType) {
+        return workType;
+      } else {
+        return null;
+      }
+    })
+    .filter(Boolean);
```

Comment:
Is there a reason we're not taking this from the searchParams (URL) as we discussed?

```js
// Id's probably find a better way of an id lookup, but it get's the point across
const workTypeFilters => searchParams.map(workTypeId => allWorkTypes.find(workType => workType.data.id === workTypeId));
```

Reply:
Yeah, this one is used for rendering the checkboxes so needs to be a combination of the aggregations returned from the API and any active filters

Reply:
The searchParams are now used for displaying the active filters

Reply:
Any chance you can leave a comment here explaining what this is doing?

Have I got it right in saying that it takes all the workTypes, and returns
1. The aggregation of that work type
2. If it's in the filter, a zero count aggregation of that filter
3. Nothing?